sqlmesh/core/schema_diff.py

```diff
@@ -337,6 +368,46 @@ def _is_coerceable_type(self, current_type: exp.DataType, new_type: exp.DataType
             return is_coerceable
         return False
 
+    def _is_precision_increase(self, current_type: exp.DataType, new_type: exp.DataType) -> bool:
+        if current_type.this == new_type.this and not current_type.is_type(
+            *exp.DataType.NESTED_TYPES
+        ):
+            current_params = self._get_type_parameters(current_type)
+            new_params = self._get_type_parameters(new_type)
+
+            if len(current_params) != len(new_params):
+                return False
+
+            return all(new >= current for current, new in zip(current_params, new_params))
+        return False
+
+    def _get_type_parameters(self, type: exp.DataType) -> t.List[t.Union[int, float]]:
+        def _str_to_number(string: str, allows_max_param: bool) -> t.Union[int, float]:
+            try:
+                return int(string)
+            except ValueError:
+                try:
+                    return float(string)
+                except ValueError:
+                    if allows_max_param and string.upper() == "MAX":
+                        return self.max_parameter_length[type.this]
+                    raise ValueError(f"Could not convert '{string}' to a number")
+
+        # extract existing parameters
+        params = [
+            _str_to_number(param.this.this, type.this in self.max_parameter_length)
+            for param in type.expressions
+        ]
+
+        # maybe get default parameter values
+        param_defaults: t.Tuple[t.Union[int, float], ...] = ()
+        if type.this in self.parameterized_type_defaults:
+            param_defaults_list = self.parameterized_type_defaults[type.this]
+            if len(params) < len(param_defaults_list):
+                param_defaults = param_defaults_list[len(params)]
```

Comment:
should this be param_defaults_list[len(params):], a splice?

how does this work?

Reply:
The index returns a tuple with all the parameters missing from a type, given the number of parameters it already has: 

```
# Bigquery's decimal parameters 
# Note that second number in first tuple 9 is different from the second tuple 0 even though they both correspond to the decimal's scale parameter
bq_decimal_params = [(38, 9), (0,)]

# "DECIMAL" has 0 params so indexes to (38,9) -> DECIMAL(38,9)
# "DECIMAL(1)" has 1 param so indexes to (0) -> DECIMAL(1,0)
# "DECIMAL(1,0)" has 2 params, so len(2) !< len(bq_decimal_params) is a no-op
```

The splice approach is cleaner, but it would only work if the second number was constant with respect to the number of parameters the provided type already has.

This approach feels opaque and inelegant to me - any ideas for a cleaner implementation?